File: University/BMMS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {

            long n = scanner.nextLong();
            long m = scanner.nextLong();

            System.out.print(gcd(n, m));
        }
    }

    public static long gcd(long p, long q) {
        if (q == 0)
            return p;
        else
            return gcd(q, p % q);
    }
}


// </Ctrl+Z>

File: University/BMMS1.py
def main():
    n = int(input())
    m = int(input())

    print(gcd(n, m))

def gcd(p, q):
    if q == 0:
        return p
    else:
        return gcd(q, p % q)

if __name__ == "__main__":
    main()
